recipes/fmt/all/conanfile.py

```diff
@@ -13,8 +15,20 @@ class FmtConan(ConanFile):
     exports_sources = ["CMakeLists.txt", "patches/**"]
     generators = "cmake"
     settings = "os", "compiler", "build_type", "arch"
-    options = {"shared": [True, False], "header_only": [True, False], "fPIC": [True, False], "with_fmt_alias": [True, False]}
-    default_options = {"shared": False, "header_only": False, "fPIC": True, "with_fmt_alias": False}
+    options = {
+        "header_only": [True, False],
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_fmt_alias": [True, False],
+        "with_os": [True, False],
+    }
+    default_options = {
+        "header_only": False,
+        "shared": False,
+        "fPIC": True,
+        "with_fmt_alias": False,
+        "with_os": True,
```

Comment:
Maybe disable the option in ´config_options´ when os not in Windows, Linux, FreeBSD, or not tools.is_apple_os?

Reply:
I'm not sure how likely it is, but I can imagine use cases where someone may not want the OS support compiled in to the library when targeting Windows, Linux, FreeBSD, etc. One such reason is minimizing the size of the binary.